### ISSUE TITLE
daily builds no longer supported / built?

### ISSUE THREAD

**Issue by bjompen** [PowerShell/PowerShell#24566]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Daily builds, as refered to in [install-powershell.ps1](https://github.com/PowerShell/PowerShell/blob/master/tools/install-powershell.ps1), does not longer seem to be built or work. Getting the metadata from [`https://aka.ms/pwsh-buildinfo-daily` ](https://github.com/PowerShell/PowerShell/blob/31b70ecd9fa2a245768f267eae2fb96a0a5ab92d/tools/install-powershell.ps1#L269) tells us that the latest daily is from March 2024, and tagged `v7.5.0-daily20240306.2`.
Not only is this old, The [download link for it](https://powershellinfraartifacts-gkhedzdeaghdezhr.z01.azurefd.net/install/v7.5.0-daily20240306.2/PowerShell-7.5.0-daily20240306.2-win-x64.zip) doesn't work.. (feel free to replace windows with other versions..)

If there is no more daily builds this should be removed from the installer, otherwise it should be fixed with new metadata / download links.

![image](https://github.com/user-attachments/assets/01664124-875e-433a-9d6c-52cea86be547)

### Expected behavior

```console
Download of daily fails
```

### Actual behavior

```console
Download shouldn't fail...
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.6
PSEdition                      Core
GitCommitId                    7.4.6
OS                             Microsoft Windows 10.0.22631
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0ΓÇª}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by ItzLevvie** [comment]
As an alternative to daily builds — use the per-commit builds from https://dev.azure.com/powershell/PowerShell/_build?definitionId=97&statusFilter=succeeded&branchFilter=37%2C37%2C37%2C37%2C37%2C37%2C37%2C37%2C37%2C37%2C37%2C37%2C37

**Comment by doctordns** [comment]
I find it only the daily build is affected.

**Comment by bjompen** [comment]
Only the daily builds are affected, yes.
Thank you for the workaround, @ItzLevvie, It is much appreciated, but the main issue is that the installer that is linked elsewhere and here still allows the selection of Daily. This may lead to confusion or errors in best case, and accidental use of really old unsupported builds in worst case.

**Comment by doctordns** [comment]
Will the daily build issue get fixed or are we no longer distributing daily builds?

**Comment by iSazonov** [comment]
@TravisEz13 Could you please comment?

**Comment by andyleejordan** [comment]
Or @adityapatwardhan?

**Comment by iSazonov** [comment]
It always seemed to me that the team should be most interested in ensuring that all partner teams continuously test on a daily basis.
(I used these builds to analyze issues when it is not possible to compile locally and quickly.)
Although we don't use daily .Net builds for testing that way (but we should).

**Comment by doctordns** [comment]
An official statement withdrawing daily builds should be made. Or how about setting the Azure artifacts available tonthw public.

**Comment by JustinGrote** [comment]
I would argue that you have the ability to do builds on *every commit* for free with github actions, and so replace the daily builds with instructions on how to fetch the latest GHA build artifact.

**Comment by bjompen** [comment]
Agreed with all of the above. Fixing an action / DevOps pipeline release / changing to per commits, they are all doable, but just like @doctordns says I believe the key is an official statement. Workarounds are only useful if we know they wont break suddenly.

**Comment by andyleejordan** [comment]
Yup, I agree with Justin. If a GitHub Action could build an _unofficial, for testing purposes only_ binary, that would be the best alternative here.

**Comment by TravisEz13** [comment]
> As an alternative to daily builds — use the per-commit builds from https://dev.azure.com/powershell/PowerShell/_build?definitionId=97&statusFilter=succeeded&branchFilter=37%2C37%2C37%2C37%2C37%2C37%2C37%2C37%2C37%2C37%2C37%2C37%2C37

@JustinGrote this should be converted to a GitHub action.    It would be better if we only pushed to a GitHub package feed daily.  

We should make it clear that these are *only* for testing.  They should not be installed in any permanent way (such as adding it to the path) in any production system, client, or server.

**Comment by doctordns** [comment]
@TravisEz13 I should probably know this - but how does that URL help me to find a daily build? 

And it would be useful to fix Install-powershell.ps1 to reflect this and to enable daily builds.

And when/why was the decision taken to stop doing daily builds

**Comment by kilasuit** [comment]
@andyleejordan sorry but it needs to be a signed package so that #24956 is not happening (as this breaks being able to run in highly locked down environments), which if being built anywhere, needs to be from this repo & not from a external source, just like it is for the preview, stable and LTS builds.

I agree with @TravisEz13 on pushing it to a GitHub package feed, as by doing so you aren't then marking it as a "release" and can caveat that only packages marked as a release & as such are available from https://github.com/PowerShell/PowerShell/releases come with full support, any other builds (daily, PR, or specific commit) have little or no official support, unless there are critical issues like security - which I expect should fall under the usual support path.

It was frustrating when the daily build stopped being available & I stopped asking team members directly about the lack of one as I had slowed down some of my testing around the time it was last made available & whilst perhaps I should have raised an issue like this but oh well we are here now. 

Without one we are as a community slowed down from not only being able to easily vaildate multiple bugs have been fixed at a time, but also much slower in being able to find any new bugs along the way. It also wastes other compute resources by having this functionality outside the repo.

I had a lovely workflow & was able to make use of the daily build side by side with supported builds including with the vscode ext & some reg edits so `pwshd` or `pwshp` or `pwshl` could be used which enabled me to open up the daily/preview/lts versions for testing things side by side very quickly & it would be great to get back to being able to use that workflow again in future.

**Comment by andyleejordan** [comment]
> [@andyleejordan](https://github.com/andyleejordan) sorry but it needs to be a signed package so that [#24956](https://github.com/PowerShell/PowerShell/issues/24956) is not happening (as this breaks being able to run in highly locked down environments), which if being built anywhere, needs to be from this repo & not from a external source, just like it is for the preview, stable and LTS builds.

A daily / dev build should _never_ be run in "highly locked down environments" as their intended purposes is for developer testing.
